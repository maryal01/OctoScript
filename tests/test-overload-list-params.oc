fn overload simulated_get(list<int> l) -> int {
    return 0;
}

fn overload simulated_get(list<bool> b) -> bool {
    return true;
}

fn overload simulated_get(list<float> f) -> float {
    return 9831.0;
}

fn overload simulated_get(list<string> s) -> string {
    return "hello";
}


print("%d %d %f %s", simulated_get([42]), simulated_get([false]), simulated_get([1.0]), simulated_get(["s"]));