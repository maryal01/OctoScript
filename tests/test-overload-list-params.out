0 1 9831.000000 hello